src/component.cpp

```diff
@@ -282,4 +282,59 @@ bool Component::hasReset(const ResetPtr &reset) const
     return mPimpl->findReset(reset) != mPimpl->mResets.end();
 }
 
+size_t getVariableIndexInComponent(const std::shared_ptr<const Component> &component, const VariablePtr &variable)
```

Comment:
Is this different to the `getVariableIndexInComponent` in `utilities.cpp` ? (is this the `shared_from_this` issues again?)

Reply:
Yes,

Reply:
because the method is `const`, `shared_from_this` returns a `std::shared_ptr<const *>` object.  As it should, unfortunately.  We probably could do with adding a `const` form of our shared pointers I guess?

Reply:
Maybe worth trying as a separate issue.

Reply:
thanks for the explanation and yes to trying in a separate issue!